src/commands/deploy/index.js

```diff
@@ -98,67 +98,61 @@ export default async ctx => {
     }
   }
 
-  const file = highlight('now.json');
-  const prop = code('version');
+  const versions = [ 1, 2 ];
+  const fileText = highlight('now.json');
+  const versionText = code('version');
+  const versionsText = code(versions.join(', '));
 
-  if (!versionFlag && !localConfig && process.stdout.isTTY) {
+  if (versionFlag) {
+    if (!versions.includes(versionFlag)) {
+      output.error(
+        `The ${param('--platform-version')} option can only be one of the following: ${versionsText}.`
+      );
+
+      return 1;
+    }
+
+    platformVersion = versionFlag;
+  } else if (!localConfig) {
     if (isFile) {
-      output.error(`There was an issue parsing ${file}`);
+      output.error(`There was an issue parsing ${fileText}`);
       return 1;
     }
 
-    output.debug(`${file} not found, generating...`);
-    const { config } = await generateProject(paths[0], output);
+    if (process.stdout.isTTY) {
+      output.debug(`${fileText} not found, generating...`);
+      const { config } = await generateProject(paths[0], output);
+
+      if (!await promptBool('Would you like to deploy?', { defaultValue: true })) return 0;
```

Comment:
I'm not sure we need to prompt here anymore. I think I originally thought we were adding a new command like `now create` but since this line gets hit from running `now`, we can assume the user is expecting to deploy. What do you think @leo ?